rancher/distros-test-framework
# Sonobuoy Conformance Results

#### Proposed Changes ####



Adding the ability to run sonobuoy and get the results from the run of conformance results



the command to run it exists within the makefile and it has been edited to accommodate the long timeout window that conformance results frequently hit within go/ginkgo expectations.

#### Types of Changes ####



It can be a quick test by altering the launchSonobuoyTests("quick") but the default is to run the full conformance suite from upstream. Primarily to be used during patch validation runs but it could be used more frequently in other ways to quickly check a clusters status. Although if you have a cluster running the fastest and easiest way to run conformance is to do it the simple way upstream does it with a simple yaml deployment. 

```
---
apiVersion: v1
kind: Namespace
metadata:
  name: conformance
---
apiVersion: v1
kind: ServiceAccount
metadata:
  labels:
    component: conformance
  name: conformance-serviceaccount
  namespace: conformance
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  labels:
    component: conformance
  name: conformance-serviceaccount-role
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: conformance-serviceaccount
subjects:
- kind: ServiceAccount
  name: conformance-serviceaccount
  namespace: conformance
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  labels:
    component: conformance
  name: conformance-serviceaccount
rules:
- apiGroups:
  - '*'
  resources:
  - '*'
  verbs:
  - '*'
- nonResourceURLs:
  - '/metrics'
  - '/logs'
  - '/logs/*'
  verbs:
  - 'get'
---
apiVersion: v1
kind: Pod
metadata:
  name: e2e-conformance-test
  namespace: conformance
spec:
  containers:
  - name: conformance-container
    image: registry.k8s.io/conformance-amd64:v1.31.2 #EDIT THIS FOR YOUR CLUSTER
    imagePullPolicy: IfNotPresent
    env:
    - name: E2E_FOCUS
      value: "\\[Conformance\\]"
    - name: E2E_SKIP
      value: ""
    - name: E2E_PROVIDER
      value: "skeleton"
    - name: E2E_PARALLEL
      value: "false"
    - name: E2E_VERBOSITY
      value: "4"
    volumeMounts:
    - name: output-volume
      mountPath: /tmp/results
  volumes:
  - name: output-volume
    hostPath:
      path: /tmp/results
  restartPolicy: Never
  serviceAccountName: conformance-serviceaccount
```

#### Testing ####


Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.
 not yet 

2. Verify code lint; we should not have errors.


3. Update the documentation if needed.
✅ 
4. Update makefile and docker run if adding new tests.
✅ 
5. Run your tests at least 4 times with all configurations needed and possible.
✅ 
6. If needed test with different os types.
✅ 
#### Linked Issues ####



#### Further Comments ####
